cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -259,7 +259,8 @@ func (client *autoscalingGceClientV1) DeleteInstances(migRef GceRef, instances [
 		req.Instances = append(req.Instances, GenerateInstanceUrl(i))
 	}
 	op, err := client.gceService.InstanceGroupManagers.DeleteInstances(migRef.Project, migRef.Zone, migRef.Name, &req).Do()
```

Comment:
What if there was a conflict deleting one instance, but not the others?

Reply:
Hmmm . . . in my mind a "delete" request is idempotent so if there is currently a delete in progress then there is no reason to consider it an error.

So delete conflicts now become successes.

If you have N successes and N delete conflicts you now have 2N successes.

If one of those N requests had an error besides a conflict error you still return that.

Reply:
I guess at this point it's an issue with the API itself?

If the API is only returning a single error that could be very problematic.

Maybe you were thinking what if there was a 500 error, but we wait for the op to finish because we think there was only a 409 error?

Reply:
https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroupManagers/deleteInstances

> Specifies whether the request should proceed despite the inclusion of instances that are not members of the group or that are already in the process of being deleted or abandoned. If this field is set to false and such an instance is specified in the request, the operation fails. The operation always fails if the request contains a malformed instance URL or a reference to an instance that exists in a zone or region other than the group's zone or region.

Reply:
@x13n I've made an alternative PR https://github.com/kubernetes/autoscaler/pull/5213 which just sets the `skipInstancesOnValidationError` parameter. I think that would correct your concern here.